src/Stryker.Core/Stryker.Core/ProjectComponents/IFileLeaf.cs

```diff
@@ -1,15 +1,19 @@
 namespace Stryker.Core.ProjectComponents
 {
+    public interface IReadOnlyFileLeaf : IReadOnlyProjectComponent
+    {
+        string SourceCode { get; }
+    }
+
     public interface IFileLeaf<T> : IFileLeaf
     {
         T SyntaxTree { get; set; }
 
         T MutatedSyntaxTree { get; set; }
     }
-    public interface IFileLeaf : IProjectComponent
-    {
-        string SourceCode { get; set; }
 
-        ReadOnlyFileLeaf ToReadOnly();
+    public interface IFileLeaf : IReadOnlyFileLeaf
+    {
+        new string SourceCode { get; set; }
```

Comment:
Is the new necessary here if you're only adding a set?

Reply:
Yes, it explicitly overrides the property